### DISCUSSION TITLE
Handle Multiple API calls in getStaticProps

### DISCUSSION THREAD

**Discussion by prasanthbattula** [vercel/next.js#28322] [Help] [answered]
I am creating dynamic route **/mutual-funds/[fund]**. For each mutual fund I have to perform 15 API calls to create a page for that mutual fund. For SEO optimization I am trying to generate static pages for some mutual funds and remaining will be generating with Incremental Static Regeneration. But due to asynchronous API calls I think site building is taking long time. On an average for one mutual fund static page generation time is 15 seconds (From Vercel Deployment).

How can I call these APIs simultaneously?

Please see below code for reference.

`export async function getStaticPaths() {
    const slugsList = await fetch(`http://localhost:8000/slugslist`)
    const slugsListResult = await slugsList.json()
    const slugs = slugsListResult["slugs"]
  
    return {
      fallback: "blocking",
      paths: slugs.slice(0, 11),
    }
}
  
  export async function getStaticProps(context) {
    const fund = context.params.fund;
  
    try {
      const avggReturnsRes = await fetch(
        `http://localhost:8000/avggaut/?name=${fund}`
      );
      var avggReturnsResult = await avggReturnsRes.json();
    } catch (error) {
      var avggReturnsResult = {};
    }
  
    try {
      const navPlotRes = await fetch(
        `http://localhost:8000/navplot/?name=${fund}`
      );
      var navPlotResult = await navPlotRes.json();
    } catch (error) {
      var navPlotResult = {};
    }
  
    try {
      const aumYearRes = await fetch(
        `http://localhost:8000/aumyearly?name=${fund}`
      );
      var aumYearResult = await aumYearRes.json();
    } catch (error) {
      var aumYearResult = {};
    }
    try {
      const fundInfoRes = await fetch(
        `http://localhost:8000/funddetails?name=${fund}`
      );
      var fundInfoResult = await fundInfoRes.json();
    } catch (error) {
      var fundInfoResult = {};
    }
    try {
      const fundInfoExtraRes = await fetch(
        `http://localhost:8000/fundinformation?name=${fund}`
      );
      var fundInfoExtraResult = await fundInfoExtraRes.json();
    } catch (error) {
      var fundInfoExtraResult = {};
    }
    try {
      const stocksRes = await fetch(
        `http://localhost:8000/stocks?name=${fund}`
      );
      var stocksResult = await stocksRes.json();
    } catch (error) {
      var stocksResult = {};
    }
    try {
      const fundInfoVRRes = await fetch(
        `http://localhost:8000/fundvr?name=${fund}`
      );
      var fundInfoVRResult = await fundInfoVRRes.json();
    } catch (error) {
      var fundInfoVRResult = {};
    }
    try {
      const yearlyyReturnsRes = await fetch(
        `http://localhost:8000/yearlyyauto/?name=${fund}`
      );
      var yearlyyReturnsResult = await yearlyyReturnsRes.json();
    } catch (error) {
      var yearlyyReturnsResult = {};
    }
    try {
      const peerCompareRes = await fetch(
        `http://localhost:8000/avgpeer?name=${fund}`
      );
      var peerCompareResult = await peerCompareRes.json();
    } catch (error) {
      var peerCompareResult = {};
    }
    try {
      const rollingYearlyRes = await fetch(
        `http://localhost:8000/rolyear?name=${fund}`
      );
      var rollingYearlyResult = await rollingYearlyRes.json();
    } catch (error) {
      var rollingYearlyResult = {};
    }
    try {
      const allRollingRes = await fetch(
        `http://localhost:8000/allrolling/?name=${fund}`
      );
      var allRollingResult = await allRollingRes.json();
    } catch (error) {
      var allRollingResult = {};
    }
    try {
      const regDirRes = await fetch(
        `http://localhost:8000/regdir/?name=${fund}`
      );
      var regDirResult = await regDirRes.json();
    } catch (error) {
      var regDirResult = {};
    }
    try {
      const bestWorstRes = await fetch(
        `http://localhost:8000/bestworst?name=${fund}`
      );
      var bestWorstResult = await bestWorstRes.json();
    } catch (error) {
      var bestWorstResult = {};
    }
    try {
      const benchmark = "Select";
      const riskParammsRes = await fetch(
        `http://localhost:8000/riskparamm/?name=${fund}&benchmark=${benchmark}`
      );
      var riskParammsResult = await riskParammsRes.json();
    } catch (error) {
      var riskParammsResult = {};
    }
  
  
  
    return {
      props: {
        fund: fund,
        titleFund: fund,
        avggReturnsResult: avggReturnsResult,
        navPlotResult: navPlotResult,
        aumYearResult: aumYearResult,
        fundInfoResult: fundInfoResult,
        fundInfoExtraResult: fundInfoExtraResult,
        stocksResult: stocksResult,
        fundInfoVRResult: fundInfoVRResult,
        yearlyyReturnsResult: yearlyyReturnsResult,
        peerCompareResult: peerCompareResult,
        rollingYearlyResult: rollingYearlyResult,
        allRollingResult: allRollingResult,
        regDirResult: regDirResult,
        bestWorstResult: bestWorstResult,
        riskParammsResult: riskParammsResult,
        
      },
      revalidate: 60,
    };
}`

**Comment by qnxdev** [comment, marked as answer]
return a Promise, and use Promise.all
